Sources/Sentry/SentryUIPerformanceTracker.m

```diff
@@ -0,0 +1,258 @@
+            [SentryUIPerformanceTracker swizzleViewControllerSubClass:[self class]];
+            return SentrySWCallOriginal(coder);
+        }),
+        SentrySwizzleModeOncePerClassAndSuperclasses, swizzleViewControllerInitWithCoder);
+
+    static const void *swizzleViewControllerInitWithNib = &swizzleViewControllerInitWithNib;
+    SEL nibSelector = NSSelectorFromString(@"initWithNibName:bundle:");
+    SentrySwizzleInstanceMethod(UIViewController.class, nibSelector, SentrySWReturnType(id),
+        SentrySWArguments(NSString * nibName, NSBundle * bundle), SentrySWReplacement({
+            [SentryUIPerformanceTracker swizzleViewControllerSubClass:[self class]];
+            return SentrySWCallOriginal(nibName, bundle);
+        }),
+        SentrySwizzleModeOncePerClassAndSuperclasses, swizzleViewControllerInitWithNib);
+#    pragma clang diagnostic pop
+#endif
+}
+
++ (void)swizzleViewControllerSubClass:(Class)class
+{
+#if SENTRY_HAS_UIKIT
+    // Swizzling only classes from the user app module to avoid track every UIKit view controller
+    // interaction.
+    static const char *appImage = nil;
+    if (appImage == nil) {
+        if ([UIApplication respondsToSelector:@selector(sharedApplication)]) {
+            UIApplication *app = [UIApplication performSelector:@selector(sharedApplication)];
+            appImage = class_getImageName(app.delegate.class);
+        }
+    }
+    if (appImage == nil || strcmp(appImage, class_getImageName(class)) != 0)
+        return;
+
+    // This are the five main functions related to UI creation in a view controller.
+    // We are swizzling it to track anything that happens inside one of this functions.
+    [SentryUIPerformanceTracker swizzleViewLayoutSubViews:class];
+    [SentryUIPerformanceTracker swizzleLoadView:class];
+    [SentryUIPerformanceTracker swizzleViewDidLoad:class];
+    [SentryUIPerformanceTracker swizzleViewWillAppear:class];
+    [SentryUIPerformanceTracker swizzleViewDidAppear:class];
+#endif
+}
+
++ (void)swizzleLoadView:(Class)class
+{
+#if SENTRY_HAS_UIKIT
+    // SentrySwizzleInstanceMethod declaration shadows a local variable. The swizzling is working
+    // fine and we accept this warning.
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wshadow"
+
+    SEL selector = NSSelectorFromString(@"loadView");
+    SentrySwizzleInstanceMethod(class, selector, SentrySWReturnType(void), SentrySWArguments(),
+        SentrySWReplacement({
+            NSString *name = [UIViewControllerHelper sanitizeViewControllerName:self];
+            NSString *spanId = [SentryPerformanceTracker.shared startSpanWithName:name
+                                                                        operation:@"navigation"];
+
+            // use the viewcontroller itself to store the spanId to avoid using a global mapper.
+            objc_setAssociatedObject(self, &SENTRY_UI_PERFORMANCE_TRACKER_SPAN_ID, spanId,
```

Comment:
`h`: I rather not inject storage to the ViewController we don't control. I think a global mapper would be a cleaner way to achieve this. There are some use cases of `objc_setAssociatedObject`, but here we can avoid it easily. 
I know we use this in [`SentryScope+Private.h`](https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentryScope+Private.m), but it already caused trouble, which I can't remember exactly.